ci/tests/puppeteer/cas.js

```diff
@@ -154,20 +157,30 @@ exports.innerTexts = async (page, selector) =>
         return results;
     }, selector);
 
+/** Get full trimmed textContent of a single element. **/
 exports.textContent = async (page, selector) => {
     let element = await page.$(selector);
     let text = await page.evaluate(element => element.textContent.trim(), element);
     await this.log(`Text content for selector [${selector}] is: [${text}]`);
     return text;
 };
 
+/** Get input value of a single element. **/
 exports.inputValue = async (page, selector) => {
     const element = await page.$(selector);
     const text = await page.evaluate(element => element.value, element);
     await this.log(`Input value for selector [${selector}] is: [${text}]`);
     return text;
 };
 
+/** Get specified attribute value of a single element. **/
+exports.attrValue = async (page, selector, attribute) => {
```

Comment:
If you do add this, you'll need to go through every op that fetches an element's attribute and rework that to use this function instead. For consistency sake, we cannot have some ops use one form and some use another. It makes it that much more difficult to refactor later.

Reply:
No prob, I remove this func.